packages/bsky/tests/views/feed-view-post.test.ts

```diff
@@ -0,0 +1,336 @@
+
+    expect(sliceA).toBeDefined()
+    expect(sliceC).toBeUndefined()
+
+    await pdsAgent.api.app.bsky.graph.block.delete(
+      { repo: carol, rkey: new AtUri(block.uri).rkey },
+      sc.getHeaders(carol),
+    )
+  })
+
+  it(`[A] -> [B] -> [C] -> [D], A blocks C, viewed as C`, async () => {
+    const A = await sc.post(alice, `A`)
+    await network.processAll()
+    const B = await sc.reply(bob, A.ref, A.ref, `B`)
+    await network.processAll()
+    const C = await sc.reply(carol, A.ref, B.ref, `C`)
+    await network.processAll()
+    const D = await sc.reply(dan, A.ref, C.ref, `D`)
+    const block = await pdsAgent.api.app.bsky.graph.block.create(
+      { repo: alice },
+      { createdAt: new Date().toISOString(), subject: carol },
+      sc.getHeaders(alice),
+    )
+
+    await network.processAll()
+
+    const timeline = await agent.api.app.bsky.feed.getTimeline(
+      { limit: LIMIT },
+      {
+        headers: await network.serviceHeaders(carol),
+      },
+    )
+
+    const sliceD = timeline.data.feed.find((f) => f.post.uri === D.ref.uriStr)
+
+    expect(sliceD).toBeDefined()
+    expect(sliceD?.reply).toBeDefined()
+
+    if (!sliceD || !sliceD.reply) {
+      throw new Error('sliceD is undefined')
+    }
+
+    expect(sliceD.reply.parent.uri).toEqual(C.ref.uriStr)
+    expect(sliceD.reply.root.uri).toEqual(A.ref.uriStr)
+    expect(AppBskyFeedDefs.isPostView(sliceD.reply.parent)).toBe(true)
+    expect(AppBskyFeedDefs.isBlockedPost(sliceD.reply.root)).toBe(true)
+
+    await pdsAgent.api.app.bsky.graph.block.delete(
+      { repo: alice, rkey: new AtUri(block.uri).rkey },
+      sc.getHeaders(alice),
+    )
+  })
+
+  it(`[A] -> [B] -> [C] -> [D], A blocks C, viewed as D`, async () => {
+    const A = await sc.post(alice, `A`)
+    await network.processAll()
+    const B = await sc.reply(bob, A.ref, A.ref, `B`)
+    await network.processAll()
+    const C = await sc.reply(carol, A.ref, B.ref, `C`)
+    await network.processAll()
+    const D = await sc.reply(dan, A.ref, C.ref, `D`)
+    const block = await pdsAgent.api.app.bsky.graph.block.create(
+      { repo: alice },
+      { createdAt: new Date().toISOString(), subject: carol },
+      sc.getHeaders(alice),
+    )
+
+    await network.processAll()
+
+    const timeline = await agent.api.app.bsky.feed.getTimeline(
+      { limit: LIMIT },
+      {
+        headers: await network.serviceHeaders(dan),
+      },
+    )
+
+    const sliceD = timeline.data.feed.find((f) => f.post.uri === D.ref.uriStr)
+
+    expect(sliceD).toBeDefined()
+    expect(sliceD?.reply).toBeDefined()
+
+    if (!sliceD || !sliceD.reply) {
+      throw new Error('sliceD is undefined')
+    }
+
+    expect(sliceD.reply.parent.uri).toEqual(C.ref.uriStr)
+    expect(sliceD.reply.root.uri).toEqual(A.ref.uriStr)
+    expect(AppBskyFeedDefs.isPostView(sliceD.reply.parent)).toBe(true)
+    expect(AppBskyFeedDefs.isPostView(sliceD.reply.root)).toBe(true)
```

Comment:
This is the case `[A] -> [B] -> [C] -> [D], A blocks C, viewed as D`.

Don't we want the root (`A`) to appear blocked, so that it doesn't appear in context of the parent (`C`)?